drivers_linux.go

```diff
@@ -1,25 +1,76 @@
 package libnetwork
 
 import (
+	"strings"
+
 	"github.com/docker/libnetwork/driverapi"
 	"github.com/docker/libnetwork/drivers/bridge"
 	"github.com/docker/libnetwork/drivers/host"
 	"github.com/docker/libnetwork/drivers/null"
 	o "github.com/docker/libnetwork/drivers/overlay"
 	"github.com/docker/libnetwork/drivers/remote"
+	"github.com/docker/libnetwork/netlabel"
 )
 
-func initDrivers(dc driverapi.DriverCallback) error {
-	for _, fn := range [](func(driverapi.DriverCallback) error){
-		bridge.Init,
-		host.Init,
-		null.Init,
-		remote.Init,
-		o.Init,
-	} {
-		if err := fn(dc); err != nil {
+func initDrivers(c *controller) error {
+	config := make(map[string]interface{})
+
+	if c.validateDatastoreConfig() {
+		config[netlabel.KVProvider] = c.cfg.Datastore.Client.Provider
+		config[netlabel.KVProviderURL] = c.cfg.Datastore.Client.Address
+	}
+
+	initializers := []struct {
+		fn    func(driverapi.DriverCallback, map[string]interface{}) error
+		ntype string
+	}{
+		{bridge.Init, "bridge"},
+		{host.Init, "host"},
+		{null.Init, "null"},
+		{remote.Init, "remote"},
+		{o.Init, "overlay"},
+	}
+
+	for _, i := range initializers {
+		updateDriverConfig(c, i.ntype, config, true)
+		if err := i.fn(c, config); err != nil {
 			return err
 		}
+		updateDriverConfig(c, i.ntype, config, false)
```

Comment:
shall we remove this call ? and rather initialize `config := make(map[string]interface{})` inside the for.
that makes the code easily readable.